search-service/src/main/java/com/movie/recommender/search/model/SearchFilter.java

```diff
@@ -0,0 +1,102 @@
+package com.movie.recommender.search.model;
+
+import lombok.Data;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+
+@Data
+public class SearchFilter implements RatingFilter, DateFilter, TitleFilter, GenresFilter, CastNamesFilter, CrewNamesFilter {
+    private String movieTitle;
+    private List<String> genres;
+    private String startDate;
+    private String endDate;
+    private Double ratingStart;
+    private Double ratingEnd;
+    private Boolean isAdult;
+    private List<String> castNames;
+    private List<String> crewNames;
+    private List<String> keywords;
+
+    @Override
+    public void validateRating() {
```

Comment:
validation logic is specific to the service, it makes more sense to handle it in the service instead of the model